woocommerce/woocommerce-gutenberg-products-block
# Fix e2e tests for price filter

Some E2E tests related to the Price Filter block were failing. I have fixed the failing tests and made some other improvements. Here are the highlights of the changes:
1. **Refactored E2E Tests for Price Filter:** We have revamped our E2E tests in `price-filter.block_theme.side_effects.spec.ts`. A key highlight of this refactoring is the introduction of a new page object model, `PriceFilterPage`, which consolidates the logic for interacting with the price filter block. This includes adding the price filter block to new posts, setting the price filter range, and locating the images and products on a page.

2. **Modified Playwright Configuration:** Changes have been made to the Playwright configuration file `playwright.config.ts` to capture full-page screenshots only on failure. This helps in better understanding the failures and improves debugging efficiency.

**Impact of Changes**
- These changes will enhance the readability and maintainability of the code.
- It will also reduce the complexity of the test suite, making it easier to understand and modify.
- The improved Playwright configuration will assist in efficient debugging of test failures.

**Areas of Concern**
- Please review the changes in the `playwright.config.ts` to ensure that it aligns with our testing strategy.
- The refactored tests need to be carefully reviewed to ensure that all aspects of the previous tests are covered.
- The new `PriceFilterPage` class should be reviewed for potential improvements in the encapsulation of interactions with the price filter block.

Part of #9875 



### Testing

1. Run `npm run env:start` or `npm run env:restart`
2. Execute end-to-end tests with Playwright using the command: `npm run test:e2e-pw`.
3. Confirm that all tests complete successfully without any failures.
4. Verify that the Continuous Integration (CI) pipeline `Playwright Tests / Playwright E2E tests` runs without any errors or failures.


### WooCommerce Visibility
